plugins/opsgenie/icon_opsgenie/util/api.py

```diff
@@ -15,7 +15,9 @@ def __init__(self, api_key: str, logger=None) -> None:
     def create_alert(self, data: dict) -> dict:
         CREATE_ALERT_URL = f"{self.api_url}alerts/"
         self.validator.validate(data)
-        return self._call_api("POST", CREATE_ALERT_URL, json_data=data)
+        create_alert_reponse = self._call_api("POST", CREATE_ALERT_URL, json_data=data)
+        request_response = self._get_request_status(create_alert_reponse.get("requestId"))
```

Comment:
The above create_alert is asynchronous on opsgenie side as they respond with 202. When you try to fetch the alert Id it may or may not exists. Add retry logic. Please use exponential backoff as there is a ratelimiting in opsgenie. https://docs.opsgenie.com/docs/api-rate-limiting

Reply:
Sure. Updated in the newest commit.